services/scheduler/src/main/java/org/mule/service/scheduler/internal/RunnableFutureDecorator.java

```diff
@@ -36,23 +37,33 @@
    * Decorates the given {@code task}
    * 
    * @param task the task to be decorated
+   * @param classLoader the context {@link ClassLoader} on which the {@code task} should be executed
    * @param scheduler the owner {@link Executor} of this task
    * @param taskAsString a {@link String} representation of the task, used for logging and troubleshooting.
    * @param id a unique it for this task.
    */
-  RunnableFutureDecorator(RunnableFuture<V> task, DefaultScheduler scheduler, String taskAsString, Integer id) {
+  RunnableFutureDecorator(RunnableFuture<V> task, ClassLoader classLoader, DefaultScheduler scheduler, String taskAsString,
+                          Integer id) {
     super(id);
     this.task = task;
+    this.classLoader = classLoader;
     this.scheduler = scheduler;
     this.taskAsString = taskAsString;
   }
 
   @Override
   public void run() {
     long startTime = beforeRun();
+
+    final Thread currentThread = Thread.currentThread();
+    final ClassLoader currentClassLoader = currentThread.getContextClassLoader();
+    currentThread.setContextClassLoader(classLoader);
+
     try {
       task.run();
     } finally {
+      currentThread.setContextClassLoader(currentClassLoader);
```

Comment:
I think this should be the last thing to be done on the task, because other things can be require the same context classloader. For example, on line 68 there is a logger, if I'm not wrong the context use dto log depends on the context classloader